### PR TITLE
layers: Update to 1.4

### PR DESCRIPTION
**Pull request by arno-lunarg**
Fixes for https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8955

### CODE REVIEW

File: layers/VkLayer_khronos_validation.json.in [hunk 1]
```diff
@@ -1057,15 +1057,30 @@
...
-                    "key": "duplicate_message_limit",
-                    "env": "VK_LAYER_DUPLICATE_MESSAGE_LIMIT",
-                    "label": "Max Duplicated Messages",
-                    "description": "Maximum number of times any single validation message should be reported. Value of zero will print forever",
-                    "type": "INT",
-                    "default": 10,
-                    "range": {
-                        "min": 0
-                    }
+                    "key": "enable_message_limit",
```
Comment (spencer-lunarg) [inline layers/VkLayer_khronos_validation.json.in:1060]: what is going on here? this is not needed for 1.4

File: scripts/generate_source.py [hunk 2]
```diff
@@ -413,6 +413,8 @@ def main(argv):
         'instrumentation_ray_query_comp.cpp',
         'instrumentation_post_process_descriptor_index_comp.h',
         'instrumentation_post_process_descriptor_index_comp.cpp',
+        'feature_requirements_helper.h',
```
Comment (spencer-lunarg) [inline scripts/generate_source.py:416]: why was this added here?

File: layers/stateless/sl_instance_device.cpp [hunk 3]
```diff
@@ -667,6 +667,35 @@ bool StatelessValidation::manual_PreCallValidateCreateDevice(VkPhysicalDevice ph
...
+             VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_DEMOTE_TO_HELPER_INVOCATION_FEATURES,
+             VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_INTEGER_DOT_PRODUCT_FEATURES,
+             VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SHADER_TERMINATE_INVOCATION_FEATURES,
+             VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SUBGROUP_SIZE_CONTROL_FEATURES,
+             VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SYNCHRONIZATION_2_FEATURES,
+             VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TEXTURE_COMPRESSION_ASTC_HDR_FEATURES,
+             VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_ZERO_INITIALIZE_WORKGROUP_MEMORY_FEATURES}};
+        while (current) {
+            if (IsValueIn(current->sType, illegal_feature_structs_with_14)) {
+                skip |= LogError("NASSIGNED-VkDeviceCreateInfo-pNext-14", physicalDevice, error_obj.location,
```
Comment (spencer-lunarg) [inline layers/stateless/sl_instance_device.cpp:689]:
```suggestion
                skip |= LogError(" VUID-VkDeviceCreateInfo-pNext-10360", physicalDevice, error_obj.location,
```

File: layers/stateless/sl_instance_device.cpp [hunk 4]
```diff
@@ -667,6 +667,35 @@ bool StatelessValidation::manual_PreCallValidateCreateDevice(VkPhysicalDevice ph
         }
     }
 
+    const auto *vulkan_14_features = vku::FindStructInPNextChain<VkPhysicalDeviceVulkan14Features>(pCreateInfo->pNext);
+    if (vulkan_14_features) {
+        const VkBaseOutStructure *current = reinterpret_cast<const VkBaseOutStructure *>(pCreateInfo->pNext);
+        const std::array<VkStructureType, 13> illegal_feature_structs_with_14 = {
```
Comment (spencer-lunarg) [inline layers/stateless/sl_instance_device.cpp:673]: are you going to update the 1.2 or 1.3 logic as well? The way we do this check shouldn't be different unless there is a comment explaining why

File: scripts/generators/device_features_generator.py [hunk 5]
```diff
@@ -217,6 +217,24 @@ def generateSource(self):
                 if (api_version >= VK_API_VERSION_1_3) {
                     features->texelBufferAlignment = true;
                 }
+                if (api_version >= VK_API_VERSION_1_4) {
```
Comment (spencer-lunarg) [inline scripts/generators/device_features_generator.py:220]: this should all be removed, this is automatically setting all these features to true, this means it would be true even if it was false from the driver

File: tests/layers/VkLayer_device_profile_api.json.in [hunk 6]
```diff
@@ -4,7 +4,7 @@
         "name": "VK_LAYER_LUNARG_device_profile_api",
         "type": "GLOBAL",
         "library_path": "@JSON_LIBRARY_PATH@",
-        "api_version": "1.3.302",
+        "api_version": "1.3.290",
```
Comment (spencer-lunarg) [inline tests/layers/VkLayer_device_profile_api.json.in:7]:
```suggestion
        "api_version": "1.4.303",
```

File: tests/unit/others.cpp [hunk 7]
```diff
@@ -1281,6 +1281,7 @@ TEST_F(VkLayerTest, MissingExtensionStruct) {
     buffer_view_ci.range = VK_WHOLE_SIZE;
     buffer_view_ci.buffer = buffer.handle();
     m_errorMonitor->SetDesiredError("VUID-VkBufferViewCreateInfo-pNext-pNext");
+    m_errorMonitor->SetDesiredError("VUID-VkBufferViewCreateInfo-pNext-pNext");
```
Comment (spencer-lunarg) [inline tests/unit/others.cpp:1284]: why does this now produce a 2nd error?

Comment (arno-lunarg) [reply tests/unit/others.cpp:1284]: The same error is emitted in two different code paths, with different messages. Will look later for the reason!

File: scripts/generators/device_features_generator.py (hunk 5 above)
Comment (arno-lunarg) [reply scripts/generators/device_features_generator.py:220]: These are the features that are required and enabled by 1.4

File: layers/stateless/sl_instance_device.cpp (hunk 4 above)
Comment (arno-lunarg) [reply layers/stateless/sl_instance_device.cpp:673]: not in this PR

File: scripts/generate_source.py (hunk 2 above)
Comment (arno-lunarg) [reply scripts/generate_source.py:416]: CI clang format issues, I have 17 and CI has 14. I will fix this later

File: scripts/generators/device_features_generator.py (hunk 5 above)
Comment (spencer-lunarg) [reply scripts/generators/device_features_generator.py:220]: read the comment above in the 1.3, that was an exception

If something like `VkPhysicalDeviceVulkan14Features::pipelineRobustness` is going to slow the device, the driver might "expose support" but the app might "not enable it"

so this currently marks it as "the app enabled it" which is wrong

File: scripts/generate_source.py (hunk 2 above)
Comment (spencer-lunarg) [reply scripts/generate_source.py:416]: add a TODO then please

File: layers/stateless/sl_instance_device.cpp (hunk 4 above)
Comment (spencer-lunarg) [reply layers/stateless/sl_instance_device.cpp:673]: add a TODO please

File: scripts/generators/device_features_generator.py (hunk 5 above)
Comment (arno-lunarg) [reply scripts/generators/device_features_generator.py:220]: Ah ok!

**Review by spencer-lunarg — APPROVED** [review]
LGTM !
